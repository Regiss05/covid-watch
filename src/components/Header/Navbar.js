import React, { useContext } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGear, faMicrophone } from '@fortawesome/free-solid-svg-icons';
import Nav from 'react-bootstrap/Nav';
import AuthContext from '../../redux/auth-context';

import classes from './Navbar.module.css';

const Navbar = () => {
  const ctx = useContext(AuthContext);
  return (
    <>
      {ctx.isLoggedIn && (
        <Nav defaultActiveKey="/" className={classes.nav} as="ul">
          <Nav.Item as="li">
            <FontAwesomeIcon
              icon={faMicrophone}
              className={`${classes.faIcon} ${classes.microphone}`}
            />
          </Nav.Item>
          <Nav.Item as="li">
            <FontAwesomeIcon
              icon={faGear}
              className={`${classes.faIcon} ${classes.gear}`}
            />
          </Nav.Item>
          <Nav.Item as="li">
            <Nav.Link eventKey="link-2">
              {' '}
              <button type="button" onClick={ctx.onLogout}>Logout</button>
            </Nav.Link>
          </Nav.Item>
        </Nav>
      )}
    </>
  );
};

export default Navbar;
