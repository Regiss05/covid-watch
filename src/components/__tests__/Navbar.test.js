import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../Header/Navbar';

describe('Test presence of Navbar', () => {
  test('should render Navbar component', () => {
    const navComponent = renderer
      .create(
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>,
      )
      .toJSON();
    expect(navComponent).toMatchSnapshot();
  });
});
