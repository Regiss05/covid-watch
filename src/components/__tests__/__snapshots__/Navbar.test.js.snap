// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test presence of Navbar should render Navbar component 1`] = `null`;
